godot-core/src/engine.rs

```diff
@@ -16,6 +16,40 @@ pub use crate::gen::central::global;
 pub use crate::gen::classes::*;
 pub use crate::gen::utilities;
 
+use self::packed_scene::GenEditState;
+
+/// Extension trait for convenience functions on `PackedScene`
+pub trait PackedSceneExt {
+    /// Instantiates the scene as type `T`, panicking if not found or bad type.
+    ///
+    /// # Panics
+    /// If the scene is not type `T` or inherited.
+    fn instantiate_as<T>(&self) -> Gd<T>
+    where
+        T: GodotClass + Inherits<Node>,
+    {
+        self.try_instantiate_as::<T>().unwrap()
+    }
+
+    /// Instantiates the scene as type `T` (fallible).
+    ///
+    /// # Panics
+    /// If the scene is not type `T` or inherited.
+    fn try_instantiate_as<T>(&self) -> Option<Gd<T>>
+    where
+        T: GodotClass + Inherits<Node>;
+}
+
+impl PackedSceneExt for PackedScene {
+    fn try_instantiate_as<T>(&self) -> Option<Gd<T>>
+    where
+        T: GodotClass + Inherits<Node>,
+    {
+        self.instantiate(GenEditState::GEN_EDIT_STATE_DISABLED)
+            .map(|gd| gd.cast::<T>())
```

Comment:
This panics if `cast()` fails, violating the method contract. Use `try_cast()` instead.